### PR TITLE
[NTOS:MM] Fix PTE protection for executable CoW pages

### PR DESCRIPTION
**Pull request by tkreuzer**
Fixes Blender x64.
See https://jira.reactos.org/browse/CORE-19761

### CODE REVIEW

**Review by oleg-dubinskiy — APPROVED** [review]

File: ntoskrnl/mm/section.c [hunk 1]
```diff
@@ -1509,7 +1509,11 @@ MmAlterViewAttributes(PMMSUPPORT AddressSpace,
                  */
                 Page = MmGetPfnForProcess(Process, Address);
 
-                Protect = PAGE_READONLY;
+                /* Choose protection based on what was requested */
+                if (NewProtect == PAGE_EXECUTE_READWRITE)
+                    Protect = PAGE_EXECUTE_READ;
+                else
+                    Protect = PAGE_READONLY;
```
Comment (TAN-Gaming) [inline ntoskrnl/mm/section.c:1516]: Nitpick: Adding a empty line below or use `{}` would be better for readability.

**Review by TAN-Gaming — APPROVED** [review]

**Review by GeoB99 — APPROVED** [review]

**Review by JoachimHenze — APPROVED** [review]
Approved. Can we try to identify the guilty-rev please?

**Comment by tkreuzer** [comment]
> Approved. Can we try to identify the guilty-rev please?

96c65e9

**Review by DarkFire01 — APPROVED** [review]

File: ntoskrnl/mm/section.c (hunk 1 above)
Comment (JoachimHenze) [reply ntoskrnl/mm/section.c:1516]: resolved.

**Pull request merged**